### PR TITLE
Performance improvement. Lazy OBO

### PR DESCRIPTION
**Pull request by jmprieur**
## Purpose
Performance improvement.
Provide a way to not do an OBO when the token is recevied by the Web API, but only when it tries to acquire another token for a downstream API.

The reason is thare are web apis which do things without necessarilly caling a downstrream API. Delaying the OBO until needed improves the performance of these Web APIs

This PR is on to of MSAL 4"s

## Does this introduce a breaking change?

```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: 2. Web API now calls Microsoft Graph/TodoListService/Startup.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ public Startup(IConfiguration configuration)
         public void ConfigureServices(IServiceCollection services)
         {
             services.AddProtectWebApiWithMicrosoftIdentityPlatformV2(Configuration)
-                    .AddProtectedApiCallsWebApis(Configuration, new string[] { "user.read" })
+                    .AddProtectedApiCallsWebApis(Configuration, null /*new string[] { "user.read" }*/)
```
Comment (jmprieur) [inline 2. Web API now calls Microsoft Graph/TodoListService/Startup.cs]: If you don't specify scopes, then the OBO will be lazy

File: Microsoft.Identity.Web/Client/TokenAcquisition.cs [hunk 2]
```diff
@@ -164,7 +164,20 @@ public async Task<string> GetAccessTokenOnBehalfOfUser(HttpContext context, IEnu
 
             // Use MSAL to get the right token to call the API
             var application = BuildConfidentialClientApplication(context, context.User);
-            return await GetAccessTokenOnBehalfOfUser(application, context.User, scopes, tenant);
+
+            // Case of a lazy OBO
+            Claim jwtClaim = context.User.FindFirst("jwt");
```
Comment (jmprieur) [inline Microsoft.Identity.Web/Client/TokenAcquisition.cs]: not sure if this is great from a security point of view to add the JWT used to call the Web API as a claim?

File: 2. Web API now calls Microsoft Graph/TodoListService/Startup.cs (hunk 1 above)
Comment (jmprieur) [reply 2. Web API now calls Microsoft Graph/TodoListService/Startup.cs]: like = scopes = null  in the definition, @bgavrilMS  ?

Comment (bgavrilMS) [reply 2. Web API now calls Microsoft Graph/TodoListService/Startup.cs]: I don't understand the question @jmprieur - I was referring to having 2 methods

.AddProtectApiCallsWebApisOnDemand
.AddProtectApiCallsWebApis

Comment (jmprieur) [reply 2. Web API now calls Microsoft Graph/TodoListService/Startup.cs]: @bgavrilMS : I rather used a default parameter (scopes = null) => if the scopes are not specified, the obo is delayed until the Web API calls a downstream API (for the user)

**Pull request merged**